openwisp_controller/config/api/zerotier_service.py

```diff
@@ -0,0 +1,109 @@
+import requests
+from django.core.exceptions import ValidationError
+
+
+class ZerotierService:
+    def _get_endpoint(self, property, operation, id):
+        _API_ENDPOINTS = {
+            'network': {
+                'create': f'/controller/network/{id}______',
+                'get': f'/controller/network/{id}',
+                'update': f'/controller/network/{id}',
+                'delete': f'/controller/network/{id}',
+            }
+        }
+        return _API_ENDPOINTS.get(property).get(operation)
+
+    def __init__(self, host, token, subnet='', ip=''):
+        self.host = host
+        self.token = token
+        self.subnet = subnet and subnet.subnet
+        self.ip_address = ip and ip.ip_address
```

Comment:
We can simplify this by only allowing string values for `subnet` and `ip` argument.

Reply:
Done